### DISCUSSION TITLE
Display deprecation messages

### DISCUSSION THREAD

**Discussion by DrewFenwick** [flora-pm/flora-server#562] [Feature Requests]
Usually in Haskage when a package is deprecated there is a message from the maintainer about what the suggested alternative is.

Flora shows whether a package is deprecated but doesn't seem to display this deprecation message.

As an example see [inline-java on Hackage](https://hackage.haskell.org/package/inline-java-0.10.0) vs [inline-java on Flora](https://flora.pm/packages/@hackage/inline-java).

It would be preferable if Flora *did* display this message next to the deprecation marker to avoid creating the appearance of deprecated packages with no alternatives and save us time clicking through to hackage docs.

**Comment by tchoutri** [comment]
I think there is some confusion here. Unless something goes wrong on our side, Flora does display alternative packages when they are listed in Hackage. 

See for example [inline-java](https://hackage.haskell.org/package/inline-java) (which does not declare a successor):
![Screenshot 2024-07-16 at 15-47-16 inline-java Java interop via inline Java code in Haskell modules](https://github.com/user-attachments/assets/1b132429-54c3-4a61-bcc0-0454f6caffed)
![Screenshot 2024-07-16 at 15-47-22 @hackage_inline-java](https://github.com/user-attachments/assets/597cd46d-3118-4e72-9f8a-aa3f3e0d602e)

---

Versus [ansi-wl-pprint](https://hackage.haskell.org/package/ansi-wl-pprint) which does specify `prettyprinter` as its successor:

![Screenshot 2024-07-16 at 15-47-28 ansi-wl-pprint The Wadler_Leijen Pretty Printer for colored ANSI terminal output](https://github.com/user-attachments/assets/5af538b3-ed0d-42d1-938c-b0590e3ac591)
![Screenshot 2024-07-16 at 15-47-34 @hackage_ansi-wl-pprint](https://github.com/user-attachments/assets/137ed1b0-9f6c-447b-aaae-f487868c2bdd)

**Reply by DrewFenwick** [reply]
Ah... I see. I had not looked into what I was seeing thoroughly and made an incorrect assumption that the missing text was some sort of deprecation message maintainers were able to leave on their package when deprecating.

It turns out the message I was seeing on  `inline-java`'s hackage page:

> The hackage package has been superseded by the github repository which builds with Bazel. Please see http://github.com/tweag/inline-java#readme.

... is the description field in it's cabal file.

I assume maintainers can mark a successor *hackage* package and this is what Flora is showing in the prettyprinter case, but obviously inline-java's successor is a Bazel built github project, not a Hackage project so they cannot specify a hackage package, and they're just naming the successor in the description field instead...

So my next thought is that hey, the fact that I can't see this message anywhere on Flora means means Flora doesn't display the description field. Should it? And what other information might maintainers be putting in their cabal description field which isn't duplicated anywhere else that Flora then hides?

But obviously if that's an issue, it's not the issue initially described in the title and opening post so if you'd prefer we continue in a new discussion/issue I'm happy for you to close this one.

**Reply by tchoutri** [reply]
> I assume maintainers can mark a successor hackage package[…]

Exact.

> So my next thought is that hey, the fact that I can't see this message anywhere on Flora means means Flora doesn't display the description field

It does when no README is available, as a fallback

>  And what other information might maintainers be putting in their cabal description field which isn't duplicated anywhere else that Flora then hides?

I don't believe any other field has redundant information like Description / README. 

> But obviously if that's an issue, it's not the issue initially described in the title and opening post so if you'd prefer we continue in a new discussion/issue I'm happy for you to close this one.

I'm not too sure myself where we are going with this. I hope that I have nonetheless answered your questions. :)
